src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1818,34 +1891,111 @@ private static void validateStyleModules(Set<Node> checks, Set<Node> configs,
-                                    + moduleName + "' should have matching " + configName + " url")
-                        .that(configUrl)
-                        .isEqualTo(expectedUrl);
+                            + moduleName + "' should have matching " + configName + " url")
+                            .that(configUrl)
+                            .isEqualTo(expectedUrl);
                 }
                 else if ("test".equals(configName)) {
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have matching " + configName + " url")
+                            + moduleName + "' should have matching " + configName + " url")
                             .that(configUrl)
                             .startsWith("https://github.com/checkstyle/checkstyle/"
                                     + "blob/master/src/it/java/com/" + styleName
                                     + "/checkstyle/test/");
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have matching " + configName + " url")
+                            + moduleName + "' should have matching " + configName + " url")
                             .that(configUrl)
                             .endsWith("/" + moduleName + "Test.java");
 
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have a test that exists")
+                            + moduleName + "' should have a test that exists")
                             .that(new File(configUrl.substring(53).replace('/',
-                                            File.separatorChar)).exists())
+                                    File.separatorChar)).exists())
                             .isTrue();
                 }
             }
         }
+    }
 
-        assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' has too many configs")
-                .that(itrConfigs.hasNext())
-                .isFalse();
+    private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
+          Iterator<Node> itrConfigs, Set<String> styleChecks, String styleName, String ruleName) {
+        boolean skip = !itrConfigs.hasNext();
+
+        while (itrChecks.hasNext()) {
+            final Node module = itrChecks.next();
+            final String moduleName = module.getTextContent().trim();
+            final String href = module.getAttributes().getNamedItem("href").getTextContent();
+            // until https://github.com/checkstyle/checkstyle/issues/13132
+            final boolean moduleIsConfig = href.startsWith("config_");
+            final boolean moduleIsCheck = href.startsWith("checks/");
+
+            if (!moduleIsConfig && !moduleIsCheck) {
+                skip = true;
```

Comment:
Remove this skip assignment here.

If we have checks, we must have atleast 1 config link and 1 test link. If we have no checks, we must have no configs or links.

After this if statement and before first assert set `hasChecks` to true. We are validating a check at this point, we so know we have a check. If we have no links, it will be caught during `itrConfigs.next`. If we have links but it is a test link, it will be caught during expected config link check.

Reply:
done